source/extensions/filters/http/peer_metadata/filter.cc

```diff
@@ -210,7 +210,14 @@ FilterConfig::FilterConfig(const io::istio::http::peer_metadata::Config& config,
       downstream_propagation_(
           buildPropagationMethods(config.downstream_propagation(), factory_context)),
       upstream_propagation_(
-          buildPropagationMethods(config.upstream_propagation(), factory_context)) {}
+          buildPropagationMethods(config.upstream_propagation(), factory_context)) {
+  strip_headers_ = false;
+  if (const char* env_p = std::getenv("ISTIO_META_STRIP_HEADERS")) {
```

Comment:
This should be a protobuf field IMHO for compliance with the rest of xDS. Envoy doesn't use environment variables for configuration.

Reply:
I am not sure about making this an official API, the consensus from upstream was this is a bug and we don't need to declare it in an API thus do it as an env var. cc a few others who may want to chime in.

@keithmattix @howardjohn @costinm

Reply:
Lin, this is data plane APIs, not user-facing APIs. xDS does not use environment variables normally, only in rare cases (backwards compatibility). This should be a field in xDS that's controlled by env variable in istiod.

Reply:
@kyessenov @hzxuzhonghu a POC for istiod to append a new filter_metadata field for services marked as external, and sending a new field in the XDS config.proto. PTAL, draft PR here: https://github.com/istio/istio/pull/46579

Reply:
@kyessenov @GregHanson thanks for the clarification! I had thought this is an env var user configures. Sounds like it is not!

Reply:
Chatting with Greg a bit, I don't think pod label makes sense as first pass, our users want to do this globally.